tests/components/watergate/test_config_flow.py

```diff
@@ -0,0 +1,83 @@
+"""Tests for the Watergate config flow."""
+
+from collections.abc import Generator
+
+import pytest
+from watergate_local_api import WatergateApiException
+
+from homeassistant.components.watergate.const import DOMAIN
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.const import CONF_IP_ADDRESS, CONF_WEBHOOK_ID
+from homeassistant.data_entry_flow import FlowResultType
+
+from .const import DEFAULT_DEVICE_STATE, MOCK_WEBHOOK_ID
+
+from tests.common import AsyncMock, HomeAssistant
+
+
+async def test_step_user_form(
+    hass: HomeAssistant,
+    mock_watergate_client: Generator[AsyncMock],
+    mock_webhook_id_generation: Generator[None],
+    mock_setup_entry: Generator[AsyncMock],
+    user_input: dict[str, str],
+) -> None:
+    """Test checking if registration form works end to end."""
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert CONF_IP_ADDRESS in result["data_schema"].schema
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input
+    )
+
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["title"] == "Sonic"
+    assert result["data"] == {**user_input, CONF_WEBHOOK_ID: MOCK_WEBHOOK_ID}
```

Comment:
Let's also assert the unique id here
`assert result["result"].unique_id == ...` I believe